src/Console/ConsoleOptionParser.php

```diff
@@ -576,9 +576,10 @@ public function addOptions(array $options)
      *
      * @param \Cake\Console\ConsoleInputSubcommand|string $name Name of the subcommand. Will also accept an instance of ConsoleInputSubcommand
      * @param array $options Array of params, see above.
+     * @param bool $sort Optional sorting of the subcommand
      * @return $this
      */
-    public function addSubcommand($name, array $options = [])
+    public function addSubcommand($name, array $options = [], $sort = true)
```

Comment:
I wonder if sort could/should be part of options array.
This way it would be also not lost when doing transformation (e.g. array format export), and the signature here stays simpler.

Reply:
Or a stateful aspect of the OptionParser. Right now, forgetting to pass sort=false in one subcommand will undo all of the other `true` values. That feels like undesirable behavior to me.

Reply:
@dereuromark I had first tested it this way, and it occurred to me that in this scope, the options being defined were actually for the subcommand. Since this PR relates to the parser functionality, I would have to agree with @markstory and his approach suggestion.

I will modify this to take that approach and update the PR.